crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -145,20 +155,22 @@ async fn _block_number_from_block_spec<T>(
             BlockTag::Earliest => Ok(U256::ZERO),
             BlockTag::Safe | BlockTag::Finalized => {
```

Comment:
In Hardhat, these require a validation of whether the blockchain's supported spec is post-Merge

Reply:
See the next line where I have a call to `confirm_post_merge_hardfork()` (though note that that function is just a stub right now).